### PR TITLE
swarm/services/chequebook: update chequebook for Solidity 4.3

### PR DESCRIPTION
**Pull request by nagydani**
- proper handling of failed send
- added fallback function with additional logging for top-ups
- added Solidity version pragma
- renamed suicide to selfdestruct

### CODE REVIEW

File: swarm/services/chequebook/contract/chequebook.sol [hunk 1]
```diff
@@ -28,16 +38,18 @@ contract chequebook is mortal {
         // Attempt sending the difference between the cumulative amount on the cheque
         // and the cumulative amount on the last cashed cheque to beneficiary.
         if (amount - sent[beneficiary] >= this.balance) {
```
Comment (lastperson) [inline swarm/services/chequebook/contract/chequebook.sol]: It should check for less than or equal, `<=`.

Comment (fjl) [inline swarm/services/chequebook/contract/chequebook.sol]: Thanks for migrating your comment to the new PR ;)

**Comment by fjl** [comment]
@nagydani Do you still want to get this merged?

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: ed94916cf1973d63d29a50e972e0fcde15a3b9de
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by fjl** [comment]
This is against the swarm branch. The contract has moved to contracts/chequebook and was modified there.

**Pull request closed without merging**